### PR TITLE
Aim overhaul and guns/gunmods balance

### PR DESCRIPTION
**Pull request by wwkk222208**
Aiming speed balance

#### Summary
None

#### Purpose of change
The current aiming time has the following problems:
1.The aiming speed of sights is almost ineffective, which results in the rifle scope is always the best choice, and the sights for CQB (such as laser sights) are trash.
2.The difficulty of high-precision shooting is not reflected. Whether it is a sniper rifle or a carbine, the aiming time they need is almost the same.
3.Due to the problem of setting the aiming attenuation curve, semi-automatic shooting can be accurate and high rate of fire at the same time, and the suppressing ability of automatic shooting is not reflected.
The purpose of this PR is to solve the above problems.

#### Describe the solution
1.A simple and gentle attenuation function is introduced to solve this problem, and the minimum recoil reduction is adjusted to 1 MOA.
2.The aiming speed of sights with the ‘zoom’ flag is compensated at low recoil to show that they are more suitable for sniping rather than close combat.
3.The maximum aiming speed determined by the size of the gun has been temporarily removed. Because the value setting of this function in the original code is unreasonable (for example, this function prevents the maximum aiming speed of the rifle from being increased by any means).
4.Since the modified skill can effectively increase the aiming speed, the gain of the gun skill on the firing speed has been greatly reduced to avoid numerical expansion.

#### Describe alternatives you've considered
Consider using a more reasonable function instead of hard cap to reflect the impact of gun volume (or length) of firearm on aiming.

#### Testing
Tested the precision shooting mode of M4 carbine and M14 EBR with rifle scope.
M4 carbine: unskilled shooter (Lv.0 skills) has little change in the aiming time of the first shot. For expert shooters (Lv.10 skills), the aiming time of the first shot is slightly faster. For different users, the aiming time other than the first shot has been extended.
M14 EBR with rifle scope: The unskilled shooter takes an extremely long time to aim (over 500 moves), and the aiming time for expert shooters is still longer than the original version (about 350 moves). After installing the offset sight, the aiming time of the expert shooter is similar to the original version (about 260 moves).

#### Additional context
In addition to rifles, other types of firearms have not yet completed detailed balancing and testing.

### CODE REVIEW

**Comment by FoolsGold45** [comment]
Can you explain in layman's terms the effect this has on gameplay? From testing it just appears to be an additional reason to never use scopes, since their effect on accuracy at long range is already dubious and it will now take much longer to take long range shots with a scope attached. The removal of the system that makes adding mods to firearms make them take longer to aim is a step in the right direction to make non-shoulder strap attachments worth it, but it seems this is only a temporary solution for this PR and not planned to be permanent. Do you have testing data related to the close-quarters sights that you mention? Does this PR make them any more worthwhile?

**Comment by wwkk222208** [comment]
> Can you explain in layman's terms the effect this has on gameplay? From reading your testing data it just appears to be an additional reason to never use scopes, since their effect on accuracy at long range is already dubious and it will now take much longer to take long any shot with a scoped firearm.
> 
> The removal of the system that makes adding mods to firearms make them take longer to aim is a step in the right direction to make non-shoulder strap attachments worth it, but it seems this is only a temporary solution for this PR and not planned to be permanent. Do you have testing data related to the close-quarters sights that you mention? Does this PR make them any more worthwhile?

**Comment by ISuckM8** [comment]
So from what I could tell from this PR, gun modifications are more useful now, snipers are pretty slow at sniping (Obviously) and pistols are generally better.

**Comment by Zireael07** [comment]
Why close? It's a fairly good overhaul!

**Comment by wwkk222208** [comment]
lol..
I just rewrite a new branch, and wil open a new PR soon.

**Comment by ISuckM8** [comment]
Godspeed,  lad.
